presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -139,7 +142,18 @@ public void start(HiveSplitSource splitSource)
     {
         this.hiveSplitSource = splitSource;
         for (int i = 0; i < maxPartitionBatchSize; i++) {
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            UserGroupInformation currentHdfsUser = getCurrentHdfsUser();
+            ResumableTasks.submit(executor, new HiveSplitLoaderTask(currentHdfsUser));
+        }
+    }
+
+    private UserGroupInformation getCurrentHdfsUser()
+    {
+        try {
+            return UserGroupInformation.getCurrentUser();
```

Comment:
Wouldn't this return the daemon user and not the querying user? I'd think you'd need to call HadoopAuthentication.getUserGroupInformation(queryingUser). Or have you already changed the current user at a higher level?